### DISCUSSION TITLE
SVTR Text Recognition: Zero Accuracy Issue & Need Guidance for Custom Pipeline

### DISCUSSION THREAD

**Discussion by btwitsayush** [PaddlePaddle/PaddleOCR#15293] [Q&A]
### **Description**
I'm new to PaddleOCR and currently building a custom text recognition pipeline for address labels and tracking numbers. While implementing this, I encountered multiple issues—most critically, the SVTR model always trains to 0% accuracy. I'm seeking help with:

- Resolving these training/export errors
- Getting guidance to correctly set up a custom text recognition pipeline

### **Current Issues**

1. SVTR model trains but stays at 0% accuracy
2. CRNN model attempts lead to segmentation faults
3. Exporting fails with SVTRRecResizeImg error

### **Environment**

- PaddlePaddle version: 2.6.2
- Running on: Google Colab (Linux)
- GPU: Available, CUDA enabled

### **Dataset Details**

- Total images: 50
- Training: 45
- Testing: 5
- Image format: BGR, 3x32x320
- Unique characters: 63
- Content: Address labels and tracking numbers (data is encrypted/masked for privacy)

### **Data Formats:**

**- Address Labels:**

Name XXXXX Address XXXXX Ltd.
Floor XX, Building XXX
City - XXXXXX
Phone XXXXXXXXXX Email Pincode XXXXXX

**Tracking Numbers:**

XXXXXXXXXXXXX (12–15 digit numbers)

### **Model Configuration (YAML)**

**Architecture:**
  model_type: rec
  algorithm: SVTR
**Backbone:**
  name: MobileNetV1Enhance
  scale: 0.5
**Neck:**
  name: SequenceEncoder
  encoder_type: svtr
  dims: 64
  depth: 2
  hidden_dims: 120
  use_guide: True
**Head:**
  name: CTCHead
  fc_decay: 0.00001

### **Training Results**

- Epochs: 20
- Best Epoch: 19
- Accuracy: 0.0
- Normalized Edit Distance: 0.06458
- Loss: From 32.9 to 31.0
- Learning Rate: 5e-6
- Pretrained Model: ch_PP-OCRv3_rec_train
- Batch Size: 4
- Max Text Length: 500

### **Questions**

1. Why is accuracy stuck at 0?
2. Is learning rate 5e-6 too low for fine-tuning?
3. How to fix the SVTRRecResizeImg export error?
4. What is the correct pipeline for training with a small, custom dataset?

### **Pipeline Help Needed**

**Please guide me on:**

- Preparing a custom dataset properly (format, label structure, etc.)
- Generating and validating the dict.txt (character dictionary)
- Adjusting training config for small datasets
- Full steps to set up a working custom text recognition pipeline
- Choosing the right architecture (SVTR, CRNN, etc.) for small  & large datasets with limited labeled images

### **Additional Notes**

- Dataset contains sensitive personal information, hence encrypted.
- Custom pipeline is for logistics use cases (address & tracking number recognition).
- Willing to contribute findings back to the community.

### **Request for Assistance**

1. Help me fix the SVTR training and export issues
2. Share a working config or example for small-scale text recognition training
3. Recommend best practices for datasets with private content
4. Guide me in building a robust custom recognition pipeline

/label ~bug ~text-recognition ~training ~help-wanted ~documentation
/cc @PaddleOCR-maintainers
